### ISSUE TITLE
KeyError: 'last-modified'

### ISSUE THREAD

**Issue by jhavens12** [instaloader/instaloader#188]
**Describe the bug**
KeyError when run

**To Reproduce**
/usr/local/bin/instaloader --login=XX --sessionfile=/mnt/insta/.instaloader-session --fast-update --no-metadata-json --no-captions --stories --highlights --tagged XXX

**Error messages and tracebacks**
Traceback (most recent call last):
  File "/usr/local/bin/instaloader", line 9, in <module>
    load_entry_point('instaloader==4.1', 'console_scripts', 'instaloader')()
  File "/usr/local/lib/python3.5/dist-packages/instaloader/__main__.py", line 388, in main
    storyitem_filter_str=args.storyitem_filter)
  File "/usr/local/lib/python3.5/dist-packages/instaloader/__main__.py", line 171, in _main
    download_stories, fast_update, post_filter, storyitem_filter)
  File "/usr/local/lib/python3.5/dist-packages/instaloader/instaloader.py", line 823, in download_profiles
    self.download_profilepic(profile)
  File "/usr/local/lib/python3.5/dist-packages/instaloader/instaloader.py", line 281, in download_profilepic
    date_object = datetime.strptime(anonymous_session.head(profile_pic_url).headers["Last-Modified"],
  File "/usr/local/lib/python3.5/dist-packages/requests/structures.py", line 54, in __getitem__
    return self._store[key.lower()][1]
KeyError: 'last-modified'

**Instaloader version**
4.1

**Comment by jhavens12** [comment]
Also reproduce-able with 4.1.1

**Comment by aandergr** [comment]
Does this problem still persist? I cannot reproduce it currently, neither did it occur during our daily unit tests.

**Comment by a1johnson** [comment]
I have the same problem using version 4.1.1. The error occurs while downloading of a lot of profile pics, during the download of a few profile pics the error never occurs.

**Comment by hgkmaxymus** [comment]
It happens to me too. Ignore the instagram profile theme/photos (lol), but I receive the error only when gathering this single profile of 350. It is open to everyone so it shouldn't have any issue.

> -> instaloader.exe --dirname-pattern={profile} --filename-pattern="{profile} - {shortcode}" --no-captions --no-metadata-json acrodave
> Traceback (most recent call last):
>   File "C:\Instaloader\__main__.py", line 442, in <module>
>   File "C:\Instaloader\__main__.py", line 438, in __main
>   File "C:\Instaloader\__main__.py", line 413, in main
>   File "C:\Instaloader\__main__.py", line 187, in _main
>   File "C:\Instaloader\instaloader\instaloader.py", line 903, in download_profiles
>   File "C:\Instaloader\instaloader\instaloader.py", line 296, in download_profilepic
>   File "C:\Users\Koch-\AppData\Local\Programs\Python\Python36\lib\site-packages\requests\structures.py", line 54, in __getitem__
> KeyError: 'last-modified'
> [11244] Failed to execute script __main__

**Comment by aandergr** [comment]
@hgkmaxymus Thanks for reporting and for mentioning a profile where that happens, making it possible for us to fix the bug with commit 4bc0a94e12834b35255874da034cfe37a552be58. Now, in case the last modified timestamp of the profile pic is not available, a hash is used in the profile pic filename.

Upgrade Instaloader with
```
pip3 install --upgrade git+https://github.com/instaloader/instaloader
```
